pyDigitalWaveTools/vcd/parser.py

```diff
@@ -74,7 +86,17 @@ def __init__(self):
         self.keyword_dispatch = defaultdict(
             self.parse_error, keyword_functions)
 
-        self.scope = None
+        # A root scope is used to deal with situations like
+        # ------
+        # $scope module testbench $end
+        # $var reg 3 ! x [2:0] $end
+        # $upscope $end
+        # $scope module testbench $end
+        # $var wire 8 " y [7:0] $end
+        # $upscope $end
+        # $enddefinitions $end
+        # ------
+        self.scope = VcdVarScope("root", None)
```

Comment:
self.scope is always initialized to this object and the if below does not wok as expected, should there be a `= None` ?
I am missing something?

Reply:
Sorry, I've been thinking about deleting all the `self.scope is None` processing when I add these lines, but I eventually forgot..
My plan is to delete `None` detections in `vcd_upscope`, `vcd_scope` to make the behaviour unified, and also for cases when multiple top level modules appear in the same vcd (which may not supposed to happen, though)